cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunnerTest.java

```diff
@@ -158,6 +158,10 @@ public void testMapreduceWithDynamicDatasets() throws Exception {
       .build());
     // build runtime args for app
     Map<String, String> runtimeArguments = Maps.newHashMap();
+
+    // Make sure there is only one mapper running at a time since this test has the Mapper writing
+    // to a dataset using increment and the in-memory table doesn't really support concurrent increment
```

Comment:
Should this be supported? What happens in distributed mode when readlessIncrements are not enabled?